app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,15 @@
+class SessionsController < ApplicationController
+    def create
+      @user = User.find_by(username: params[:username])
+  
+      if @user && @user.authenticate(params[:password])
+        session[:user_id] = @user.id

```

Comment:
General question for my understanding, how is this receiving the id value? The person logging in seems to pass a username and password so I don't follow how the id is being generated.

Reply:
I follow the process in "https://dev.to/kjdowns/creating-a-user-login-system-ruby-on-rails-2kl2" in log-in process

I believe if the user is successfully authenticated, this line assigns the user's ID to the :user_id key in the session hash.